### DISCUSSION TITLE
Doesn't work

### DISCUSSION THREAD

**Discussion by morrigan** [magefile/mage-action#303] [Q&A] [answered]
I've tried using this in several ways but always end up hitting the wall regarding directory setup.

1. When `buildall` is run without any flags, error is: `No .go files marked with the mage build tag in this directory.`
[Run output](https://github.com/NodeFactoryIo/hactar-daemon/runs/497656805?check_suite_focus=true)

2. with `-d` flag passed, magefile output error is: `cannot find package "github.com/magefile/mage/mg"`

[Run output](https://github.com/NodeFactoryIo/hactar-daemon/runs/498040370?check_suite_focus=true)

Is there an example repo where this is used in some other way?

**Comment by crazy-max** [comment, marked as answer]
Hi @morrigan,

> When buildall is run without any flags, error is: `No .go files marked with the mage build tag in this directory.`
> with -d flag passed, magefile output error is: cannot find package "github.com/magefile/mage/mg"

That's because you [checkout your repo in a custom path](https://github.com/NodeFactoryIo/hactar-daemon/actions/runs/52905684/workflow#L20).

As I can see, you rely on go modules so you shouldn't use `GOPATH` anyway. Fyi, `GOPATH` is deprecated since Go 1.13.

> Is there an example repo where this is used in some other way?

You can find working examples here: https://github.com/search?q=%22ghaction-mage%22&type=Code

**Comment by morrigan** [comment]
Hi @crazy-max, thanks for the answering. You are right, however I did try a few different approaches. [Here](https://github.com/NodeFactoryIo/hactar-daemon/runs/497643594?check_suite_focus=true) is one without custom path.

I've managed to solve what I need using custom mage install but I think it might help someone else by looking into it.

**Comment by crazy-max** [comment]
@morrigan Yes that's because you don't run `go mod download` in your mage file. It's dangerous because it will not retrieve deps based on your `go.mod/sum` (stick versions) but instead your `GOPATH` (and download latest from internet if not found).
You can take a look [here](https://github.com/crazy-max/WindowsSpyBlocker/blob/3ee2d528374a798f787b1acf00dc1b36cd36e0b9/magefile.go#L70-L78) to find out how to use `go mod download` in your magefile.
